crates/cast/src/starknet_commands/script.rs

```diff
@@ -0,0 +1,340 @@
+use std::any::Any;
+use std::collections::HashMap;
+use std::fs;
+
+use crate::starknet_commands::call;
+use anyhow::{anyhow, ensure, Context, Result};
+use cairo_felt::Felt252;
+use cairo_lang_casm::hints::{Hint, StarknetHint};
+use cairo_lang_casm::operand::{CellRef, ResOperand};
+use cairo_lang_runner::casm_run::{cell_ref_to_relocatable, execute_core_hint_base};
+use cairo_lang_runner::casm_run::{extract_relocatable, vm_get_range, MemBuffer};
+use cairo_lang_runner::short_string::as_cairo_short_string;
+use cairo_lang_runner::{
+    build_hints_dict, insert_value_to_cellref, RunResultValue, SierraCasmRunner,
+};
+use cairo_lang_sierra::program::VersionedProgram;
+use cairo_lang_sierra_to_casm::metadata::MetadataComputationConfig;
+use cairo_lang_utils::bigint::BigIntAsHex;
+use cairo_lang_utils::ordered_hash_map::OrderedHashMap;
+use cairo_vm::hint_processor::hint_processor_definition::{HintProcessorLogic, HintReference};
+use cairo_vm::serde::deserialize_program::ApTracking;
+use cairo_vm::types::exec_scope::ExecutionScopes;
+use cairo_vm::vm::errors::hint_errors::HintError;
+use cairo_vm::vm::errors::vm_errors::VirtualMachineError;
+use cairo_vm::vm::runners::cairo_runner::{ResourceTracker, RunResources};
+use cairo_vm::vm::vm_core::VirtualMachine;
+use camino::Utf8PathBuf;
+use cast::helpers::response_structs::ScriptResponse;
+use cast::helpers::scarb_utils::{
+    get_package_metadata, get_scarb_manifest_for, get_scarb_metadata,
+};
+use cheatnet::cheatcodes::EnhancedHintError;
+use clap::command;
+use clap::Args;
+use conversions::StarknetConversions;
+use indoc::formatdoc;
+use itertools::chain;
+use num_traits::ToPrimitive;
+use scarb_metadata::ScarbCommand;
+use starknet::core::types::{BlockId, BlockTag::Pending, FieldElement};
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::JsonRpcClient;
+use tokio::runtime::Runtime;
+
+#[derive(Args)]
+#[command(about = "")]
+pub struct Script {
+    /// Path to the script
+    pub script_path: Utf8PathBuf,
```

Comment:
I don't like this being based on paths, imo it should be workspace focused to be consistent with scarb/forge philosophy. 

Maybe better would be to have scripts package with fixed name `scripts` and call it by a funciton name?

Reply:
https://github.com/foundry-rs/starknet-foundry/pull/1067/commits/a429f89e743c6f364899f541655e5857e8c6ec37 
updated as discussed offline 

Now scripts have to be put into `scripts` folder.  To execute a script, the user will have to pass the script module name.

Reply:
Ok, now it has been changed. Scripts can exist as a separate Scarb package. We use `path_to_scarb_toml` to launch scripts from an arbitrary directory. `path_to_scarb_toml` is expected to point to the `Scarb.toml` within the package with scripts

Reply:
Context: after discussion with Maciek we decided that if we do this that way, we can still run script based on package and script name, but we can also add extrernal contracts as dependencies which we would want to do either way in the near future